### PR TITLE
Feature/kak/import local anaysis jobs#617

### LINKED ISSUE #617: Implement AnalysisJob import function
This task is to write a function in Django that takes a UUID to a AnalysisJobLocalUpload model instance and does the following:
- [ ] Set AnalysisJobLocalUpload task model to a running state
- [ ] Create a new AnalysisJob object for the model
- [ ] Push each file in the original zipfile of results to the expected S3 location for the AnalyisJob
- [ ] Copy ways and census blocks into the database for the AnalysisJob
- [ ] Mark the AnalysisJobLocalUpload task and the newly created AnalysisJob models as complete

Depends on #619

### LINKED ISSUE #620: Validate user local analysis upload zip file
This task is to write a function that takes the user uploaded zip file and verifies that the zip contains all of the expected files. A complete list, based on a local job I ran, is provided below.

Eventually we'll call this in the API POST being created in #618 before running the import function developed in #617.

```
andrew@shenandoah ~/src/pfb-network-connectivity/data/output/4d8aa837-49b3-4e6c-816b-ecc451da4181 (feature/awf/job-id-in-analysis-shapefiles): ls -lh
total 10216
-rw-r--r--  1 andrew  staff   3.4M Jun 27 11:43 neighborhood_census_blocks.geojson
-rw-r--r--  1 andrew  staff   583K Jun 27 11:43 neighborhood_census_blocks.zip
-rw-r--r--  1 andrew  staff   217B Jun 27 11:23 neighborhood_colleges.geojson
-rw-r--r--  1 andrew  staff   217B Jun 27 11:23 neighborhood_community_centers.geojson
-rw-r--r--  1 andrew  staff   353K Jun 27 11:23 neighborhood_connected_census_blocks.csv.zip
-rw-r--r--  1 andrew  staff   217B Jun 27 11:23 neighborhood_dentists.geojson
-rw-r--r--  1 andrew  staff   1.1K Jun 27 11:23 neighborhood_doctors.geojson
-rw-r--r--  1 andrew  staff   1.1K Jun 27 11:23 neighborhood_hospitals.geojson
-rw-r--r--  1 andrew  staff   2.2K Jun 27 11:23 neighborhood_overall_scores.csv
-rw-r--r--  1 andrew  staff    14K Jun 27 11:23 neighborhood_parks.geojson
-rw-r--r--  1 andrew  staff   512B Jun 27 11:23 neighborhood_pharmacies.geojson
-rw-r--r--  1 andrew  staff   217B Jun 27 11:23 neighborhood_retail.geojson
-rw-r--r--  1 andrew  staff    18K Jun 27 11:23 neighborhood_schools.geojson
-rw-r--r--  1 andrew  staff    52K Jun 27 11:23 neighborhood_score_inputs.csv
-rw-r--r--  1 andrew  staff    18K Jun 27 11:23 neighborhood_social_services.geojson
-rw-r--r--  1 andrew  staff   2.0K Jun 27 11:23 neighborhood_supermarkets.geojson
-rw-r--r--  1 andrew  staff   5.0K Jun 27 11:23 neighborhood_transit.geojson
-rw-r--r--  1 andrew  staff   2.2K Jun 27 11:23 neighborhood_universities.geojson
-rw-r--r--  1 andrew  staff   551K Jun 27 11:43 neighborhood_ways.zip
```

Depends on #617

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Extract, validate, and upload local analysis results from provided URL in task.

### Demo

A completed job:

image

Re-imported neighborhood ways and Census blocks for that job:

pfb_reimported_geo_results_console

The files uploaded for the job to S3:

image

Error message returned on attempting to upload a zip file missing results:

pfb_api_task_error_missing_files

## Testing Instructions

 * Upload local analysis results to a publicly accessible URL (or use [mine](https://s3.amazonaws.com/kat-pfb-storage-us-east-1/germantown_results.zip))
 * Create a local analysis upload task at http://localhost:9200/api/local_upload_tasks/
 * Get the job UUID from the returned upload task JSON
 * Check all 19 results file are on S3 in the results directory named like: `https://s3.console.aws.amazon.com/s3/buckets/{$USER}-pfb-storage-us-east-1/results/{JOB_ID}/`
 * Check in the Django console that Census block and neighborhood ways results exist for the job ID
 * Attempting to create a local analysis upload task with an incomplete/invalid results zip file should return an appropriate error
 * Job and upload task status should match expected stages

Closes #617 
Closes #620

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
The refactor came out a bit longer, but maybe easier to read.

_Note to self to rebase before merge._

File: src/django/pfb_analysis/tasks.py [hunk 1]
```diff
@@ -23,5 +44,86 @@ def create_batch_from_remote_shapefile(shapefile_url):
...
+        # Run the local analysis download, extract, and upload for the task
+        logging.info('Starting local analysis upload {uuid} from {url}'.format(
+                     uuid=local_upload_task_uuid, url=task.upload_results_url))
+        download_and_extract_local_results(tmpdir, task)
+        upload_and_insert_local_results(tmpdir, task)
+    except (ObjectDoesNotExist, ValidationError):
+        logging.error('No local upload analysis task found for {uuid}.'.format(
+                      uuid=local_upload_task_uuid))
+    except LocalAnalysisFetchException as ex:
+        task.status = AnalysisLocalUploadTask.Status.ERROR
```
Comment (KlaasH) [inline src/django/pfb_analysis/tasks.py]: I haven't tracked down why, but when I provide an invalid zip file, I see the "Missing expected results files" error in the log output, but the task status isn't getting changed from `CREATED`.

Comment (flibbertigibbet) [reply src/django/pfb_analysis/tasks.py]: The initial `POST` response will say `CREATED`. Does a subsequent requery still not show an error status? If so, could you share the URL to the test file?

Comment (KlaasH) [reply src/django/pfb_analysis/tasks.py]: Doh, sorry.  I was looking at the `analysis_job` list, not the `local_upload_tasks` list.  The upload task is getting the proper error status and message.

**Comment by flibbertigibbet** [comment]
Updated. Good catch on `update_status`.

**Comment by flibbertigibbet** [comment]
Modified to set overall scores, which I refactored for easier access outside of the management command.

**Pull request merged**